test/errors/incompatibleKinds-detailed.good

```diff
@@ -0,0 +1,49 @@
+─── error in incompatibleKinds.chpl:3 [IncompatibleKinds] ───
+  A regular variable cannot be initialized with a type.
+  In the following initialization expression:
+      |
+    3 | var a = int;
+      |         ⎺⎺⎺
+      |
+  the initialization expression is the type 'int(64)'.
+
+─── error in incompatibleKinds.chpl:4 [IncompatibleKinds] ───
+  A type variable cannot be initialized with a regular value.
+  In the following initialization expression:
+      |
+    4 | type b = notAType;
+      |          ⎺⎺⎺⎺⎺⎺⎺⎺
+      |
+  the initialization expression is a value of type 'string'.
+  If you were trying to extract the type of the expression on the left of the '=', try using '.type'?
+
+─── error in incompatibleKinds.chpl:7 [IncompatibleKinds] ───
+  A 'param' cannot be initialized with a non-'param' value.
+  In the following initialization expression:
+      |
+    7 | param d = c;
+      |           ⎺
+      |
+  the initialization expression is a value of type 'int(64)'.
+  Note that 'param' values must be known at compile-time.
+
+─── error in incompatibleKinds.chpl:10 [IncompatibleKinds] ───
+  A regular variable cannot be initialized with a type.
+  In the following initialization expression:
+       |
+    10 | var e = MyRecord;
+       |         ⎺⎺⎺⎺⎺⎺⎺⎺
+       |
+  the initialization expression is the type 'MyRecord'.
+  If you're trying to create a new value of type 'MyRecord', try writing 'new MyRecord()' instead
+
+─── error in incompatibleKinds.chpl:13 [IncompatibleKinds] ───
+  A regular variable cannot be initialized with a type.
+  In the following initialization expression:
+       |
+    13 | var f = aliasToRecord;
+       |         ⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺
+       |
+  the initialization expression is the type 'MyRecord'.
```

Comment:
I see the discussion at https://github.com/chapel-lang/chapel/pull/20776/files#r998782829; I think even if we are not able to report all the information we would want here, we should still mention that it is going through an alias. Perhaps `the initialization expression is an alias of the type 'MyRecord'`.

Reply:
At this time, we don't include aliasing information. Furthermore, this PR is already quite bloated in my opinion, so I will open an issue for this.

Reply:
Don't you have enough information to say that it's an alias [here](https://github.com/DanilaFe/chapel/blob/better-errors-2/frontend/lib/framework/error-classes-list.cpp#L253)?

Either way, it can certainly be saved for another issue.

Reply:
For a composite type, yes; not all types have names, though. A more principled approach would require "external" support from Dyno / QualifiedType / etc.

Reply:
I'm not snapping to it:  What's an example of an expression that would be a type and be legal in this syntactic position?  I guess maybe something like `[1..10] real`?

Reply:
Functions can return types, so practically any arbitrary (compile-time) expression can be valid in this syntactic position, as long as the outermost return intent is `type`.

Reply:
So in that case would we say "but the initialization expression is the type 'theTypeTheFunctionReturns'" or would we say something more like "but the initialization expression is a type expression" or "but the initialization expression is the type procedure call 'myTypeRoutine()'".  To me, the first feels odd, at least without some wording changes.